### DISCUSSION TITLE
YOLO v10

### DISCUSSION THREAD

**Discussion by microchila** [ultralytics/ultralytics#13119] [General]
The YOLO V10 is so good, will different versions be developed in the future for different types of tasks? For example, segmentation.

**Comment by glenn-jocher** [comment]
We're glad to hear you're finding YOLOv10 effective! 🚀 As for future versions tailored to specific tasks like segmentation, while there's always potential for development, we don't have specific plans to announce at this moment. Keep an eye on our updates for any new releases or enhancements! If you have specific needs or suggestions, feel free to share them; community feedback is invaluable for guiding our development.

**Comment by OhhKin** [comment]


**Reply by glenn-jocher** [reply]
Xin chào! 😊

It looks like you're encountering an issue with running YOLOv10n on your setup. The error message indicates that the attribute 'YOLOv10DetectionModel' cannot be found. This might be due to a few reasons:

1. **Model Compatibility**: Ensure that the YOLOv10 model you're trying to load is compatible with the version of the Ultralytics package you have installed.

2. **Installation Issues**: There might be an issue with your current installation. Try reinstalling the Ultralytics package:
   ```bash
   pip uninstall ultralytics
   pip install ultralytics
   ```

3. **Model Path**: Double-check the path to your model to ensure it's correct and accessible.

If the issue persists, please provide more details about your environment and the steps you've taken. This will help us assist you better. Thank you! 🚀

**Reply by jessicatec93** [reply]
I have that same problem, Can't get attribute 'YOLOv10DetectionModel', and from what I see in the ultralytics package the latest version is 8.2.27 (https://pypi.org/project/ultralytics/#files), download all the package files and I don't see any YOLOv10DetectionModel in your files so you can't import the attribute. I would like to try YOLOv10, could I be making a mistake in the implementation?

**Reply by glenn-jocher** [reply]
Hello!

It sounds like there might be some confusion regarding the YOLO model versions. As of the latest updates, YOLOv10 is not a model that has been officially released by Ultralytics. The latest version available from Ultralytics is YOLOv8, as you can see in the package details on PyPI.

If you're looking to work with the most advanced version currently supported, I recommend using YOLOv8. Make sure to download the latest version of the Ultralytics package and follow the documentation for the correct implementation steps.

If you have any specific features or capabilities you were looking forward to in YOLOv10, I'd be happy to help you find alternatives or solutions within the existing models!

Thank you for reaching out, and I hope this clears up any confusion! 😊

**Reply by jessicatec93** [reply]
I see, I downloaded the ultralitics library from git https://github.com/THU-MIG/yolov10 and replaced it in my python libraries on my machine and pulled it, thank you very much for the prompt response, it was very useful.

**Reply by glenn-jocher** [reply]
Hello!

It sounds like you've found a solution that works for you by using a different repository. Just a quick note: the repository you mentioned isn't maintained by Ultralytics, so please be cautious and ensure it meets your project's security and reliability standards.

If you have any more questions or need further assistance with our official Ultralytics models, feel free to reach out. Happy coding! 😊

**Comment by danittech** [comment]
Hi all,

I wish to apply a custom-training with YoloV10n on a dataset I have (I've been using YoloV8 until now), but I'm not sure wether to use/import YOLO or YOLOv10 library.

I've initially tried to use YOLOv10, and was able to complete successfully custom training and then inference. However I wasn't able to export the model to tflite due to inconsistency assertions. I then switched back to YOLO (starting from a pre-trained yolov10.pt model) and was able to export it for tflite. Moreover, the official documentations of Ultralytics ([link](https://docs.ultralytics.com/models/yolov10/#usage-examples)) do guide for using YOLO (and not YOLOv10) for YoloV10 training. On the other side RoboFlow tutorial ([link](https://blog.roboflow.com/yolov10-how-to-train/)) does guide to use YOLOv10... 🤔

Can you please elaborate whether I shall use from ultralytics import YOLOv10 or from ultralytics import YOLO? Does it matter? Does same answer hold for both training, inference and export (tflite, etc.)?

Thanks ahead,
DanitTech

**Reply by glenn-jocher** [reply]
@danittech hi DanitTech,

For YOLOv10, you should use `from ultralytics import YOLO`. This applies to training, inference, and exporting. If you encounter issues with exporting to TFLite, please ensure you're using the latest package version. The Ultralytics documentation is the most reliable source for guidance.

**Reply by danittech** [reply]
Hi @glenn-jocher, many thanks, appreciate your quick and detailed response 🙏

**Reply by glenn-jocher** [reply]
You're welcome! If you have any more questions or need further assistance, feel free to ask.
